pkg/util/protomarshal/protomarshal.go

```diff
@@ -52,15 +53,16 @@ func UnmarshalAllowUnknown(b []byte, m proto.Message) error {
 	return unmarshaler.Unmarshal(bytes.NewReader(b), legacyproto.MessageV1(m))
 }
 
-func UnmarshalAllowUnknownWithAnyResolver(anyResolver jsonpb.AnyResolver, b []byte, m proto.Message) error {
-	return (&jsonpb.Unmarshaler{
-		AllowUnknownFields: true,
-		AnyResolver:        anyResolver,
-	}).Unmarshal(bytes.NewReader(b), legacyproto.MessageV1(m))
+type resolver interface {
+	protoregistry.MessageTypeResolver
+	protoregistry.ExtensionTypeResolver
 }
 
-func UnmarshalWithGlobalTypesResolver(b []byte, m proto.Message) error {
-	return protojson.Unmarshal(b, m)
+func UnmarshalAllowUnknownWithAnyResolver(anyResolver resolver, b []byte, m proto.Message) error {
+	return (&protojson.UnmarshalOptions{
```

Comment:
Why are we switching to protojson? The new json library has toxic behavior of injecting random spaces in the output; we intentionally use the 'legacy' jsonpb to get sane behavior. Do we need to change?

Reply:
I guess this is unmarshal

Reply:
I was trying to use jsonpb but failed to handle nested Any types for the wasm config, where it showed unmarshal errors like `Any JSON doesn't have '@type'`. Do you have any examples to do it with the jsonpb?

In searching for a workable solution, I saw that the `proxy-config ecds` uses protojson to handle ECDS config unmarshalling, therefore I switched to protojson, and added the similar logic to handle the unknown fields.